File: apps/agor-android/app/src/main/kotlin/live/agor/app/models/MCPServer.kt
package live.agor.app.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class MCPServer(
    @SerialName("mcp_server_id") val mcpServerId: String,
    val name: String,
    val description: String? = null,
    val transport: String? = null,
    val url: String? = null,
    val command: String? = null,
    val args: List<String>? = null,
    @SerialName("oauth_authenticated") val oauthAuthenticated: Boolean? = null,
    @SerialName("requires_oauth") val requiresOauth: Boolean? = null,
)

@Serializable
data class SessionMCPServer(
    @SerialName("session_id") val sessionId: String,
    @SerialName("mcp_server_id") val mcpServerId: String,
    val enabled: Boolean = true,
    @SerialName("added_at") val addedAt: String? = null,
)

@Serializable
data class ServerProfile(
    val id: String,
    val label: String,
    val url: String,
    val email: String? = null,
    @SerialName("default") val isDefault: Boolean = false,
)

File: apps/agor-rust-android/crates/agor/src/models/mcp_server.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct MCPServer {
    pub mcp_server_id: String,
    pub name: String,
    pub description: Option<String>,
    pub transport: Option<String>,
    pub url: Option<String>,
    pub command: Option<String>,
    pub args: Option<Vec<String>>,
    pub oauth_authenticated: Option<bool>,
    pub requires_oauth: Option<bool>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct SessionMCPServer {
    pub session_id: String,
    pub mcp_server_id: String,
    pub enabled: bool,
    pub added_at: Option<String>,
}
